python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -93,6 +110,108 @@ def serialize(self) -> str:
         return json.dumps(storage_dict)
 
 
+def execute_asset_backfill_iteration(
+    backfill: "PartitionBackfill", workspace: BaseWorkspaceRequestContext, instance: DagsterInstance
+) -> None:
+    """
+    Runs an iteration of the backfill, including submitting runs and updating the backfill object
+    in the DB.
+    """
+    from dagster._core.execution.backfill import BulkActionStatus
+
+    asset_graph = ExternalAssetGraph.from_workspace_request_context(workspace)
+    if backfill.serialized_asset_backfill_data is None:
+        check.failed("Asset backfill missing serialized_asset_backfill_data")
+
+    asset_backfill_data = AssetBackfillData.from_serialized(
+        backfill.serialized_asset_backfill_data, asset_graph
+    )
+
+    result = execute_asset_backfill_iteration_inner(
```

Comment:
I think potentially you'd also want to change this to an iterator to allow for heartbeating, if this is potentially going to take a long time to execute.

Reply:
👍